File: q8.py
def q8(): 
    a = 1
    print("Starting a = " + str(a))
    def sub1():
            a = 8
            b = 2
            print(" Sub 1: a = " + str(a) + ", b = " + str(b))
            def sub2():
                b = 24
                c = 3
                print(" Sub 2: a = " + str(a) + ", b = " + str(b) + ", c = " + str(c))
                def sub3():
                    c = 81
                    d = 4
                    print(" Sub 3: a = " + str(a) + ", b = " + str(b) + ", c = " + str(c) + ", d = " + str(d))
                sub3()
            sub2()
    sub1()
q8()

File: q8.js
<html>
<title>Web Page Design</title>
<head>
<script>
function q8() {
  var a = 1;
  console.log("Starting a = " + a);
  function sub1(){
      var a = 8;
      var b = 2;
      console.log(" Sub 1: a = " + a + ", b = " + b);
      
      function sub2(){
         var b = 24;
         var c = 3;
         console.log(" Sub 2: a = " + a + ", b = " + b + ", c = " + c);
         
         function sub3(){
             var c = 81;
             var d = 4;
             console.log(" Sub 3: a = " + a + ", b = " + b + ", c = " + c + ", d = " + d);
         }
         sub3();
      }
      sub2();
  }
  sub1();
}
q8();
</script>
</head>
<body>
</body>
</html>